backend/scrapers/em2.js

```diff
@@ -0,0 +1,371 @@
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });
+    this.skipFurtherChecks.push(event.title);
+    return {
+      event,
+      reason: reasons.reverse().join(','),
+      success: true,
+    };
+  }
+
+  this.talkToDB({
+    type: 'db-request',
+    subtype: 'isRefused',
+    messageData: {
+      string: event.title,
+    },
+  });
+  await this.checkDBhasAnswered();
+  reasons.push(this.lastDBAnswer.reason);
+  if (this.lastDBAnswer.success) {
+    return {
+      event,
+      reason: reasons.reverse().join(','),
+      success: false,
+    };
+  }
+
+  const hasForbiddenTerms = await this.hasForbiddenTerms(event);
+  reasons.push(hasForbiddenTerms.reason);
+  if (hasForbiddenTerms.success) {
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveRefusedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });    
+    
+    return {
+      event,
+      reason: reasons.reverse().join(','),
+      success: false,
+    };
+  }
+
+  const isRockRes = await this.isRock(event);
+  if (isRockRes.success) {
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    }); 
+    return {
+      event,
+      reason: reasons.reverse().join(', '),
+      success: true,
+    };
+  }
+
+  const goodTermsRes = await this.hasGoodTerms(event);
+  reasons.push(goodTermsRes.reason);
+  if (goodTermsRes.success) {
+    this.skipFurtherChecks.push(event.title);
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });
+    return goodTermsRes;
+  }
+
+
+  this.talkToDB({
+    type: 'db-request',
+    subtype: 'saveRefusedTitle',
+    messageData: {
+      string: event.title,
+      reason: reasons.reverse().join(', '),
+    },
+  });   
+  return {
+    event,
+    reason: reasons.reverse().join(', '),
+    success: false,
+  };
+};
+// #endregion                          MAIN PAGE EVENT CHECK
+
+// #region [rgba(0, 180, 0, 0.1)]      SINGLE PAGE EVENT CHECK
+// #endregion                          SINGLE PAGE EVENT CHECK
+
+// #region [rgba(0, 240, 0, 0.1)]      MAIN PAGE
+scraper.mainPage = async function () {
+  const availableBaseEvents = await this.checkBaseEventAvailable(workerData.family);
+  if (availableBaseEvents) {
+    const thisWorkersEvents = availableBaseEvents.filter(
+      (eventEl, index) => index % workerData.workerCount === workerData.index,
+    );
+    return this.mainPageEnd({ stopFunctie: null, rawEvents: thisWorkersEvents });
+  }
+
+  const { stopFunctie, page } = await this.mainPageStart();
+
+  await this.waitTime(50);
+
```

Comment:
Je zou hier waitForSelector moeten gebruiken. Nu kijk je in essentie op page load of die selector bestaat terwijl het ruikt naar een dynamische waarde